src/Spago/Messages.hs

```diff
@@ -228,3 +228,7 @@ globsDoNotMatchWhenWatching patterns = makeMessage $
 
 makeMessage :: [Text] -> Text
 makeMessage = Text.intercalate "\n"
+
+sourceImportsTransitiveDependency :: [Text] -> Text
+sourceImportsTransitiveDependency transitive =
+  "Source files import from transative dependency: " <> Text.intercalate ", " transitive
```

Comment:
This error message could definitely use some work - I just put this up as a first pass to get people thinking about it.

Reply:
Maybe something like this?
```
Some of your project files import modules from packages that are not in the direct dependencies of your project.
To fix this error add the following packages to the list of dependencies in your config:
- ...
```